Volume/Views/MainView/TabContainer.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  TabViewContainer.swift
+//  Volume
+//
+//  Created by Sergio Diaz on 4/10/21.
+//  Copyright © 2021 Cornell AppDev. All rights reserved.
+//
+
+import SwiftUI
+
+struct TabContainer<Content: View>: View {
+    @EnvironmentObject private var networkState: NetworkState
+    let content: Content
+
+    init(content: () -> Content) {
+        self.content = content()
+    }
+
+    var body: some View {
+        ZStack {
+            self.content
+
+            if networkState.networkFailed {
+                ConnectionFailedView()
+                    .offset(x: 0, y: 48) // Offset includes navigation bar
```

Comment:
The navigation bar has a different heigh on different devices. Is there a more extensible way to do this?

Reply:
48 was actually something I calculated experimentally that worked universally on all devices.  I believe the reason the navigation bar looks like it has more height on the iPhone 12, for example, is actually because of the safe edges.  That's why I believe this 48 works.  However, I do agree it doesn't look too good and if for some reason the size of the navigation bar changed, the UI would be messed up.  I can look into another way of doing this.